polkadot/node/core/approval-voting/src/time.rs

```diff
@@ -88,3 +97,157 @@ pub(crate) fn slot_number_to_tick(slot_duration_millis: u64, slot: Slot) -> Tick
 	let ticks_per_slot = slot_duration_millis / TICK_DURATION_MILLIS;
 	u64::from(slot) * ticks_per_slot
 }
+
+/// A list of delayed futures that gets triggered when the waiting time has expired and it is
```

Comment:
I would move this tracking inside the `BlockEntry` and also include ` candidates_pending_signature` inside the new structu. Polling it should return a `Vec` of candidates  that are to be signed now.

Reply:
BlockEntry is actually reflecting a structure in the database, we don't seem to be holding in memory all active BlockEntries, the way I see it this is just keeping in memory a list of hashes that we know have candidates waiting for signatures to be created which we then load from the database. This is similar with the CurrentlyCheckingSet.

I don't see how we can put this futures on the BlockEntry structure, because that would force us to keep in the currently working memory all block entries that need attention, could you describe in more detail what you had in mind here ?